search/sublist_search.cpp

```diff
@@ -0,0 +1,338 @@
+/**
+ * @file
+ * @brief Implementation of the [Sublist Search
+ * Algorithm](https://www.geeksforgeeks.org/sublist-search-search-a-linked-list-in-another-list)
+ * @details
+ *  * Sublist search is used to detect a presence of one list in another list.
+ *  * Suppose we have a single-node list (let's say the first list), and we
+ *  want to ensure that the list is present in another list (let's say the
+ * second list), then we can perform the sublist search to find it.
+ *
+ *  * For instance, the first list contains these elements: 23 -> 30 -> 41,
+ *  and the second list contains these elements: 10 -> 15 -> 23 -> 30 -> 41
+ *  -> 49. At a glance, we see that the first list presents in the second list.
+ *
+ * ### Working
+ *
+ *  * The sublist search algorithm works by comparing the first element
+ *  of the first list with the first element of the second list.
+ *  * If the two values don't match, it goes to the next element of the
+ *  second list. It does this until the two values match.
+ *
+ * @author [Nitin Sharma](https://github.com/foo290)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace search
+ * @brief Searching algorithms
+ */
+namespace search {
+/**
+ * @namespace sublist_search
+ * @brief Functions for the [Sublist
+ * Search](https://www.geeksforgeeks.org/sublist-search-search-a-linked-list-in-another-list)
+ * implementation
+ */
+namespace sublist_search {
+/**
+ * @brief A Node structure representing a single link Node in a linked list
+ */
+struct Node {
+    uint32_t data = 0;
+    Node *next{};
```

Comment:
Add a one-line description of what these variables are for (see the example below).
```c++
int64_t data = 0;       ///< The value/key of the node
```